examples/routing_service/routing_service_msql_adapter/cpp/ConnectorBase.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+ *
+ * RTI grants Licensee a license to use, modify, compile, and create derivative
+ * works of the Software.  Licensee has the right to distribute object form
+ * only for use with RTI products.  The Software is provided "as is", with no
+ * warranty of any type, including any warranty for fitness for any purpose.
+ * RTI is under no obligation to maintain or support the Software.  RTI shall
+ * not be liable for any incidental or consequential damages arising out of the
+ * use or inability to use the software.
+ */
+
+#ifndef CONNECTOR_BASE_HPP_
+#define CONNECTOR_BASE_HPP_
+
+#include <dds/core/xtypes/DynamicData.hpp>
+
+/*
+ * Abstract Base class with implementations for connect() and disconnect()
+ */
+class ConnectorBase {
+public:
+    ConnectorBase(const std::string topicName);
+    virtual ~ConnectorBase() = 0;
+
+    /* default implementations */
+    virtual auto connect() -> bool;
+    virtual void disconnect();
+    virtual auto connected() -> bool;
+
+    /* must be implemented by derived connectors */
+    virtual auto writeData(const dds::core::xtypes::DynamicData *sample)
```

Comment:
Perhaps this is a big change now, but we use snake_case for method_names and variables in modern C++.

Reply:
I can make the changes but this will take some time.

Reply:
Don't worry about that, we'll do it after the merge if necessary.